contracts/SwapKiwi.sol

```diff
@@ -2,285 +2,321 @@
-
-    delete _swaps[swapId];
-  }
-
-  /**
-    * @dev Returns NFTs from SwapKiwi to swap initator.
-    *      Callable only if second user hasn't yet added NFTs.
-    *
-    * @param swapId ID of the swap that the swap participants want to cancel
-    */
-  function cancelSwap(uint256 swapId) external {
-    require(
-      _swaps[swapId].initiator == msg.sender || _swaps[swapId].secondUser == msg.sender,
-      "SwapKiwi: Can't cancel swap, must be swap participant"
-    );
-      // return initiator NFTs
-      safeMultipleTransfersFrom(
-        address(this),
-        _swaps[swapId].initiator,
-        _swaps[swapId].initiatorNftAddresses,
-        _swaps[swapId].initiatorNftIds
-      );
-
-    if(_swaps[swapId].secondUserNftAddresses.length != 0) {
-      // return second user NFTs
-      safeMultipleTransfersFrom(
-        address(this),
-        _swaps[swapId].secondUser,
-        _swaps[swapId].secondUserNftAddresses,
-        _swaps[swapId].secondUserNftIds
-      );
-    }
-
-    if (_swaps[swapId].initiatorEtherValue != 0) {
-      _etherLocked -= _swaps[swapId].initiatorEtherValue;
-      uint amountToTransfer = _swaps[swapId].initiatorEtherValue;
-      _swaps[swapId].initiatorEtherValue = 0;
-      _swaps[swapId].initiator.transfer(amountToTransfer);
-    }
-    if (_swaps[swapId].secondUserEtherValue != 0) {
-      _etherLocked -= _swaps[swapId].secondUserEtherValue;
-      uint amountToTransfer = _swaps[swapId].secondUserEtherValue;
-      _swaps[swapId].secondUserEtherValue = 0;
-      _swaps[swapId].secondUser.transfer(amountToTransfer);
-    }
-
-    emit SwapCanceled(msg.sender, swapId);
-
-    delete _swaps[swapId];
-  }
-
-  function safeMultipleTransfersFrom(
-      address from,
-      address to,
-      address[] memory nftAddresses,
-      uint256[] memory nftIds
-    ) internal virtual {
-    for (uint256 i=0; i < nftIds.length; i++){
-      safeTransferFrom(from, to, nftAddresses[i], nftIds[i], "");
-    }
-  }
-
-  function safeTransferFrom(
-      address from,
-      address to,
-      address tokenAddress,
-      uint256 tokenId,
-      bytes memory _data
-    ) internal virtual {
-    IERC721(tokenAddress).safeTransferFrom(from, to, tokenId, _data);
-  }
-
-  function withdrawEther(address payable recipient) external onlyOwner {
-    require(recipient != address(0), "SwapKiwi: transfer to the zero address");
-
-    recipient.transfer((address(this).balance - _etherLocked));
-  }
-
-  function onERC721Received(
-    /* solhint-disable */
-      address operator,
-      address from,
-      uint256 tokenId,
-      bytes calldata data
-    /* solhint-enable */
-    ) external pure override returns (bytes4) {
-      return bytes4(keccak256("onERC721Received(address,address,uint256,bytes)"));
-  }
+/**
+* @title This is the contract which added erc1155 into the previous swap contract.
+*/
+contract SwapKiwi is Ownable, ERC721Holder, ERC1155Holder {
+
+	uint64 private _swapsCounter;
```

Comment:
I actually wonder if it's cheaper to leave at uin256. I read somewhere that if it isn't packed with other stuff it would cost more gas